evalml/utils/woodwork_utils.py

```diff
@@ -79,14 +97,17 @@ def infer_feature_types(data, feature_types=None):
                 ww_error = f"{ww_error}. Please initialize ww with df.ww.init() to get rid of this message."
             raise ValueError(ww_error)
         data.ww.init(schema=data.ww.schema)
-        return data
+        return convert_all_nan_unknown_to_double(data)
 
     if isinstance(data, pd.Series):
+        if all([isinstance(x, type(pd.NA)) for x in data]):
```

Comment:
Can we use the vectorized form of this, e.g. `(x == pd.NA).all()`, here and in `is_column_pd_na` ?